aderyn_core/src/detect/low/reverts_and_requries_in_loops.rs

```diff
@@ -0,0 +1,103 @@
+use std::collections::BTreeMap;
+use std::error::Error;
+
+use crate::ast::{NodeID, NodeType};
+
+use crate::capture;
+use crate::context::browser::GetClosestAncestorOfTypeX;
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct RevertsAndRequiresInLoopsDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for RevertsAndRequiresInLoopsDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        // Collect all require statements
+        let requires_and_reverts = context
+            .identifiers()
+            .into_iter()
+            .filter(|&id| id.name == "revert" || id.name == "require")
+            .collect::<Vec<_>>();
+
+        for item in requires_and_reverts {
```

Comment:
@TilakMaddy an interesting future benchmark will be to test this retrieval method, vs:
1. Get all loops
2. Extract reverts and requires from them